app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/EventsViewModel.java

```diff
@@ -13,21 +14,23 @@
 public class EventsViewModel extends ViewModel {
 
     private final EventRepository model;
-    private final MutableLiveData<List<Event>> events;
-
-    //TODO add event list filtering.
+    private final MutableLiveData<List<Event>> events = new MutableLiveData<>(new ArrayList<>());
+    private EventFilter filter = EventFilter.NONE;
 
     public EventsViewModel(EventRepository model) {
         this.model = model;
-        events = new MutableLiveData<>(new ArrayList<>());
         refreshEvents();
     }
 
     public LiveData<List<Event>> getEventListLiveData() {
         return events;
     }
 
+    public void setFilter(EventFilter filter) {
```

Comment:
Have you thought of another way than using a setter here?

Reply:
I could pass the filter every time on refresh but I found that it complicates the code and since there is only one instance of the view model the lack of immutability is not drastic
KISS